### ISSUE TITLE
Extremely slow execution time for a simple query

### ISSUE THREAD

**Issue by Parkyprg** [editfmah/sharkorm#40]
I really like this framework and I am struggling to make it work flawless in my application. 
But I am encountering a few issues and I don't if it is something that I am doing wrong or not.

For example, I have a table "Contact" which basically stores AddressBook contacts.
It's definition is as follows:
@property NSString      *phoneNumber;
@property NSString      *formatedPhoneNumber;
@property NSString      *firstName;
@property NSString      *lastName;
@property NSString      *compositeName;
@property NSString      *phoneLabel;
@property NSString      *profilePictureUrl;
@property NSNumber      *recordID;
@property BOOL          isFriend;
@property BOOL          isPrivate;
@property BOOL          checked;
@property BOOL          smsSent;
@property int               source;
@property double        dateAdded;
@property double        lastModified;

There is one index added, on 'source' field:
[idx addIndexForProperty:@"source" propertyOrder:SRKIndexSortOrderAscending];

Currently there are 3010 rows in this table.
If I run this:
_[[[Contact query] where:@"source in (1, 2)"] fetch];_
I have the following results:

```
[SQL query] SELECT Contact.source as result$source, Contact.lastName as result$lastName, Contact.Id as result$Id, Contact.firstName as result$firstName, Contact.recordID as result$recordID, Contact.formatedPhoneNumber as result$formatedPhoneNumber, Contact.smsSent as result$smsSent, Contact.phoneLabel as result$phoneLabel, Contact.phoneNumber as result$phoneNumber, Contact.checked as result$checked, Contact.compositeName as result$compositeName, Contact.isFriend as result$isFriend, Contact.isPrivate as result$isPrivate, Contact.lastModified as result$lastModified, Contact.dateAdded as result$dateAdded, Contact.profilePictureUrl as result$profilePictureUrl FROM Contact WHERE source in (1, 2)   
[ Compiled query ] (null)
[ Execution plan ] (
        {
        detail = "SEARCH TABLE Contact USING INDEX idx_Contact_source_asc (source=?)";
        from = 0;
        order = 0;
        selectid = 0;
    },
        {
        detail = "EXECUTE LIST SUBQUERY 1";
        from = 0;
        order = 0;
        selectid = 0;
    }
)
Rows: 0, parseTime: 0, firstResultTime: 0, queryTime: 3489, lockObtainTime: 0
```

Why it takes 3.5 seconds to get the results? Running on a DB viewer, with the same data, it gives the results immediately.
There are no other queries that are running in the same time, so there is no lock incurred.

If I get all the rows, the situation is not getting too much better either:
_[[Contact query] fetch];_

```
[SQL query] SELECT Contact.source as result$source, Contact.lastName as result$lastName, Contact.Id as result$Id, Contact.firstName as result$firstName, Contact.recordID as result$recordID, Contact.formatedPhoneNumber as result$formatedPhoneNumber, Contact.smsSent as result$smsSent, Contact.phoneLabel as result$phoneLabel, Contact.phoneNumber as result$phoneNumber, Contact.checked as result$checked, Contact.compositeName as result$compositeName, Contact.isFriend as result$isFriend, Contact.isPrivate as result$isPrivate, Contact.lastModified as result$lastModified, Contact.dateAdded as result$dateAdded, Contact.profilePictureUrl as result$profilePictureUrl FROM Contact    
[Compiled query] (null)
[Execution plan] (
        {
        detail = "SCAN TABLE Contact";
        from = 0;
        order = 0;
        selectid = 0;
    }
)
Rows: 0, parseTime: 0, firstResultTime: 0, queryTime: 2777, lockObtainTime: 0
```

Tests were done on a iPhone 5s and as I mentioned before - 3010 rows are in this table.
Will try later with a simple table, with only 1 column and check the results.

Thank you for taking time looking at this.

**Comment by editfmah** [comment]
Yes that is extremely slow, and certainly not what I would ever expect.  It's good to see that it is using the index, but how an indexed search could ever be slower than a table scan I have no idea.

Interesting though, is the **Rows:0** part, because in that last query it should have returned all 3k of the records.

Can you see if a **fetchLightweight** is faster?  or the speed of the same query except using a call to **Ids**.

Basically what i'm looking for is.  Is SQLite performing the query slowly, or is the creation of the class objects taking the time.

It could be that my latest changes have had a profound effect on the speed of the framework that I had not noticed and that were not picked up in the tests.

**Comment by Parkyprg** [comment]
I just tried on an empty project and I have better results (the same number of rows):

`Rows: 0, parseTime: 0, firstResultTime: 0, queryTime: 811, lockObtainTime: 0`

But this makes me wonder why in the other project the time is 3-4 times bigger?

**Comment by Parkyprg** [comment]
Ah - and still getting Rows 0 in the new project.

**Comment by editfmah** [comment]
Okay, i'm looking now.
